### PR TITLE
KEYCLOAK-5182 Port event-listener-sysout and event-store-mem to QS

### PR DESCRIPTION
**Pull request by martin-kanis**

### CODE REVIEW

File: event-listener-sysout/src/test/java/org/keycloak/quickstart/event/listener/ArquillianSysoutEventListenerProviderTest.java [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+    private void removeSpi() throws IOException, CliException, InterruptedException {
+        ModelNodeResult result = onlineManagementClient.execute("/subsystem=keycloak-server/spi=eventsListener/:remove");
+        result.assertSuccess();
+        result = onlineManagementClient.execute("reload");
+        TimeUnit.SECONDS.sleep(5);
+        result.assertSuccess();
+    }
+
+    private void checkServerLogForEvents(String log, EventType... events) {
+        Arrays.stream(events).map(event -> log.contains(event.name())).forEach(Assert::assertTrue);
```
Comment (hmlnarik) [inline event-listener-sysout/src/test/java/org/keycloak/quickstart/event/listener/ArquillianSysoutEventListenerProviderTest.java]: This actually tests nothing.

**Comment by martin-kanis** [comment]
PR is updated

**Comment by martin-kanis** [comment]
@abstractj added

**Comment by hmlnarik** [comment]
Hitting https://issues.jboss.org/projects/WFMP/issues/WFMP-94 when trying to run the tests from command line on JDK 10, 8u141, 8u131.

File: event-store-mem/README.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+Target Product: Keycloak  
+Source: <https://github.com/keycloak/keycloak-quickstarts>
+
+What is it?
+-----------
+
+This example shows how to implement and deploy an event store provider that stores events in memory.
+You can extend the server's functionality in this way by implementing your own provider. For more information see our server developer guide.
+
+After the provider is deployed and enabled in Keycloak, log out and log in back. Go to [Login Events](http://localhost:8180/auth/admin/master/console/#/realms/master/events) tab in the admin console where you can see login events showed from the Event store provider.
```
Comment (hmlnarik) [inline event-store-mem/README.md]: This needs to detail how to enable the event provider in Keycloak manually. https://github.com/keycloak/keycloak/blob/master/examples/providers/event-store-mem/README.md

**Comment by hmlnarik** [comment]
Just tested both QS manually and they work (when using -Dwildfly.skip=true and updated pom.xml to overcome WFMP-94). So once automation is fixed, I believe this one would be ready to go.

**Comment by martin-kanis** [comment]
@hmlnarik @abstractj  rebased and updated

**Comment by abstractj** [comment]
@martin-kanis @hmlnarik we need this to be merged first https://github.com/keycloak/keycloak-quickstarts/pull/107. Otherwise, the build will always be broken.

File: event-store-mem/README.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+What is it?
+-----------
+
+This example shows how to implement and deploy an event store provider that stores events in memory.
+You can extend the server's functionality in this way by implementing your own provider. For more information see our server developer guide.
+
+After the provider is deployed and enabled in Keycloak, log out and log in back. To enable the event provider manually go to [Events Config](http://localhost:8180/auth/admin/master/console/#/realms/master/events-settings) tab in the admin console and enable login events and admin events by a toggle buttons. You can configure what event types should be stored. Save the changes afterwards. Go to [Login Events](http://localhost:8180/auth/admin/master/console/#/realms/master/events) tab in the admin console where you can see login events showed from the Event store provider.
+Similarly in [Admin Events](http://localhost:8180/auth/admin/master/console/#/realms/master/admin-events) tab you can see generated admin events. To generate an admin event you can for example create an user.
+
+
```
Comment (hmlnarik) [inline event-store-mem/README.md]: This still needs to detail how to enable the event provider in Keycloak manually. Currently configuration updates are missing. https://github.com/keycloak/keycloak/blob/master/examples/providers/event-store-mem/README.md

File: event-store-mem/README.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+System Requirements
+-------------------
+
+You need to have Keycloak running with an initial user created. The example requires that Kecloak server is running on port 8180 (management port 10090). See the parent [README](https://github.com/keycloak/keycloak-quickstarts#start-the-keycloak-server) for more details.
```
Comment (hmlnarik) [inline event-store-mem/README.md]: Kecloak -> Keycloak

**Comment by martin-kanis** [comment]
I've improved event-store-mem tests and README and rebased to the latest master.

**Comment by abstractj** [comment]
@martin-kanis Hynek found something at the logs [here](https://api.travis-ci.org/v3/job/367624119/log.txt). Did you see the errors? Seems like there's something wrong with the tests:

```
  phantomjs://platform/console++.js:263 in error
[ERROR - 2018-04-17T13:39:42.762Z] Session [c2441050-4244-11e8-909b-474627ed388e] - page.onError - msg: Possibly unhandled rejection: {"data":"<html><head><title>Error</title></head><body>404 - Not Found</body></html>","status":404,"config":{"method":"GET","transformRequest":[null,null],"transformResponse":[null],"jsonpCallbackParam":"callback","url":"/auth/admin/realms","headers":{"Accept":"application/json, text/plain, */*","Authorization":"Bearer eyJhbGciOiJSUzI1NiIsInR5cCIgOiAiSldUIiwia2lkIiA6ICJGSjg2R2NGM2pUYk5MT2NvNE52WmtVQ0lVbWZZQ3FvcXRPUWVNZmJoTmxFIn0.eyJqdGkiOiI4MTBhZWNkZi00ZTk5LTQwMmItYTM0NC1jNzk1OGZmNWExNDciLCJleHAiOjE1MjM5NzI0NDIsIm5iZiI6MCwiaWF0IjoxNTIzOTcyMzgyLCJpc3MiOiJodHRwOi8vMTI3LjAuMC4xOjgxODAvYXV0aC9yZWFsbXMvZXZlbnQtc3RvcmUtbWVtIiwiYXVkIjoic2VjdXJpdHktYWRtaW4tY29uc29sZSIsInN1YiI6IjgzNGE1MDMwLTlkNTUtNDkyYy04OGMwLWMxMjUzMzljMDcyOSIsInR5cCI6IkJlYXJlciIsImF6cCI6InNlY3VyaXR5LWFkbWluLWNvbnNvbGUiLCJub25jZSI6IjRjYmI1YzVlLTY5ZWEtNDI0Yy04MDUwLWYzMzUxMWNlMjg5NCIsImF1dGhfdGltZSI6MTUyMzk3MjM4Miwic2Vzc2lvbl9zdGF0ZSI6IjgyOWY3ZWNiLTNiOGQtNDg1Zi1hOGU3LTY0ODU4YzhhMDQ5MyIsImFjciI6IjEiLCJhbGxvd2VkLW9yaWdpbnMiOltdLCJyZXNvdXJjZV9hY2Nlc3MiOnt9LCJuYW1lIjoiQWRtaW4gVGVzdCIsInByZWZlcnJlZF91c2VybmFtZSI6InRlc3QtYWRtaW4iLCJnaXZlbl9uYW1lIjoiQWRtaW4iLCJmYW1pbHlfbmFtZSI6IlRlc3QiLCJlbWFpbCI6InRlc3RAYWRtaW4ub3JnIn0.efkZSWn44UG94QvihZ344Gb0_7Gwdw0Eu7uTnaFhPbOCmjUsQzt0gl4-hRRebXeBsW1RU0re7LrCFz-fraD2MAVKCi0RMicHrmsged0zCzJ5ay85ukcETjzojkE9VO9t4EdIRgJsteqmqJ9YLY58bqtK_3hvGFUXmDyl2olwLUE"}},"statusText":"Not Found","xhrStatus":"complete","resource":[]}

```

**Comment by martin-kanis** [comment]
@abstractj no idea what is that. I wasn't able to find out.

**Comment by abstractj** [comment]
> @abstractj no idea what is that. I wasn't able to find out.

We need to figure this out. Otherwise, this is certainly going to haunt us in the near future.

**Comment by vmuzikar** [comment]
@martin-kanis Imho it's not a problem with the test itself nor webdriver. The test's passing, right? Seems more like a simple log message from AngularJS, i.e. caused by the Keycloak JS adapter which doesn't handle the 401 HTTP status properly (which is not necessarily a problem but still generates an error message).
PhantomJS [has some JS errors handling](http://phantomjs.org/api/webpage/handler/on-error.html) (unlike other browsers) so it looks like it just passed this browser error to the test log.

I suggest to try and reproduce the test steps manually in some proper browser and watch the browser's dev console. I'm pretty sure the same error appears there as well. In such case I'd peacefully merged this PR as it's not a test problem.

**Comment by vmuzikar** [comment]
Guys, any update on this? :) Do you need help with that?

**Comment by abstractj** [comment]
@vmuzikar I believe your help would be more than welcome. @martin-kanis can correct me if I'm wrong, but he didn't manage to reproduce the steps locally. If you can help, that would be very appreciated.

File: event-store-mem/README.md [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+    ````
+    KEYCLOAK_HOME/standalone/deployments (for Linux)
+    KEYCLOAK_HOME\standalone\deployments (for Windows)
+    ````  
+
+
+Integration test of the Quickstart
+----------------------------------
+
+1. Make sure you have a Keycloak server running with an admin user in the `master` realm
```
Comment (vmuzikar) [inline event-store-mem/README.md]: Should we mention that the `admin` user needs password set to `admin`?

File: event-listener-sysout/README.md [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    KEYCLOAK_HOME\standalone\deployments (for Windows)
+    ````  
+Then go to [Events Config](http://localhost:8180/auth/admin/master/console/#/realms/master/events-settings) tab in the admin console and add `sysout` to Event Listeners.
+Save the changes afterwards. 
+
+
+Integration test of the Quickstart
+----------------------------------
+
+1. Make sure you have a Keycloak server running with an admin user in the `master` realm
```
Comment (vmuzikar) [inline event-listener-sysout/README.md]: Should we mention that the `admin` user needs password set to `admin`?

File: event-store-mem/README.md [hunk 7]
```diff
@@ -0,0 +1,72 @@
+event-store-mem: Event Store that stores events in memory
+=========================================================
+
+Level: Beginner  
+Technologies: JavaEE  
+Summary: Example of Event Store provider
+Target Product: Keycloak
```
Comment (vmuzikar) [inline event-store-mem/README.md]: There's no line break between `Summary:` and `Target Product:`.

File: event-store-mem/README.md [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+    <spi name="eventsStore">
+        <default-provider>in-mem</default-provider>
+    </spi>
+    ````
+Then go to [Events Config](http://localhost:8180/auth/admin/master/console/#/realms/master/events-settings) tab in the admin console and enable login events and admin events by a toggle buttons. You can configure what event types should be stored. Save the changes afterwards.
+
+To apply changes in the configuration file restart the server.
+
+When you have the example deployed and configured go to [Login Events](http://localhost:8180/auth/admin/master/console/#/realms/master/events) tab in the admin console where you can see login events showed from the Event store provider.
+Similarly in [Admin Events](http://localhost:8180/auth/admin/master/console/#/realms/master/admin-events) tab you can see generated admin events. To generate an admin event you can for example create an user.
```
Comment (vmuzikar) [inline event-store-mem/README.md]: Should be "... create **a** user".

**Comment by abstractj** [comment]
@hmlnarik I think after the changes suggested by @vmuzikar. @martin-kanis can file a Jira for keycloak.js and we can merge this PR. What do you think?

**Pull request merged**